src/main/java/com/aws/greengrass/util/BatchedSubscriber.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.util;
+
+import com.aws.greengrass.config.ChildChanged;
+import com.aws.greengrass.config.Node;
+import com.aws.greengrass.config.Subscriber;
+import com.aws.greengrass.config.Topic;
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.config.WhatHappened;
+
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.BiPredicate;
+
+/**
+ * Utility class that allows a subscription to only fire
+ * on the last change in a batch.
+ *
+ * <p>This is useful in scenarios such as configuration changes
+ * where you may only want to perform expensive work once within
+ * a short span.
+ */
+public final class BatchedSubscriber implements ChildChanged, Subscriber {
+
+    private static final BiPredicate<WhatHappened, Node> BASE_EXCLUSION = (what, child) ->
+            what == WhatHappened.timestampUpdated || what == WhatHappened.interiorAdded;
+
+    private final Node node;
+    private final AtomicInteger numRequestedChanges = new AtomicInteger();
+
+    private final BiPredicate<WhatHappened, Node> exclusions;
+    private final Callback callback;
+
+    /**
+     * Callback to perform after a batch of changes fires.
+     */
+    public interface Callback {
```

Comment:
this interface exists already, it is a `Consumer<WhatHappened>`, reuse the builtin Java interface since this doesn't add anything